src/packet/vp9.rs

```diff
@@ -13,6 +13,83 @@ const MAX_VP9REF_PICS: usize = 3;
 
+/// Vp9 information describing the depacketized/packetized data.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct Vp9CodecExtra {
+    /// Map of the SVC layers.
+    ///
+    /// Index of each element corresponds to `spatial layer` index.
+    ///  
+    /// Each element represents the end (not including) in [`MediaData::data`]
+    /// for `spatial layer` it belongs.
+    ///
+    /// The all `spatial layer`s (up to 3) in [`MediaData::data`] are sorted by
+    /// `spatial layer` index from lower to greater. So, the begining of some
+    /// `spatial layer` is the end (including) of the previous `spatial layer`
+    /// (for the `spatial layer` with index 0 the begining is 0).
+    ///
+    /// ```text
+    /// [MediaData::data]:
+    /// +-----+----------------------+---------------------------+
+    /// | SL0 |         SL1          |            SL2            |
+    /// +-----+----------------------+---------------------------+
+    /// |     |                      |                           |
+    /// V     V                      V                           V
+    /// 0   layers_scheme[0]       layers_scheme[1]            layers_scheme[2]
+    /// ```
+    ///
+    /// [`MediaData::data`]: crate::media::MediaData::data
+    ///
+    /// ## Example
+    ///
+    /// Say you have a VP9 track working in "L3T3" scalability mode and you
+    /// want to retranslate S1T2 layer. So you need to retranslate the all
+    /// spatial and temporal layers <= target layer.
+    ///
+    /// ```no_run
+    /// # use str0m::{format::CodecExtra, media::MediaData};
+    /// #
+    /// // We want to receive S1T2 layer.
+    /// let (target_spatial, target_temporal) = (1u8, 2u8);
+    ///
+    /// let mut media_data: MediaData = todo!();
+    ///
+    /// if let CodecExtra::Vp9(vp9_extra) = media_data.codec_extra {
+    ///     // Try to get current `temporal layer`. If `None` there is no SVC.
+    ///     if let Some(tid) = vp9_extra.tid {
+    ///         if let (true, Some(end_of_target_layer)) = (
+    ///             // Check whether we need this `temporal layer`.
+    ///             tid <= target_temporal,
+    ///             // Check whether we have information about the right
+    ///             // border of data for the target `spatial layer`.
+    ///             vp9_extra.layers_scheme[target_spatial as usize],
+    ///         ) {
+    ///             // Cut off unwanted data of `spatial layer`s with spatial
+    ///             // indeces higher than `target_spatial`.
+    ///             media_data.data.drain(end_of_target_layer..);
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    pub layers_scheme: [Option<usize>; 3],
+
+    /// Temporal layer id.
+    pub tid: Option<u8>,
+
+    /// Map of the SVC layers widths.
+    ///
+    /// Specified for every spatial layer.
+    pub layers_widths: [Option<u16>; 3],
+
+    /// Map of the SVC layers heights.
+    ///
+    /// Specified for every spatial layer.
```

Comment:
I wonder if it would be nicer to group `layer_*` into a struct by layer

Reply:
Or to join all information specific for the layer under the one `struct LayerInfo {position: usize, width: u16, height: u16}`?

Reply:
Yes, I meant joining them. I am trying to imagine how it gets used

```
struct SpatialLayerInfo {
    start: usize, end: usize, width: u16, height: u16
}
```

One can use it like

```
let info = media_data.extra.layers.find(/* by resolution */);
let data_slice = media_data.data[info.start..info.end];
_ = decode(data_slice);
```

or

```
// remove layer 3
let info = media_data.extra.layers[3];
media_data.data.drain(info.start..);
media_data.extra.layers.drain(2..);
```

Reply:
Yes, I got your idea. I tried to do the same at the beginning. But there is a problem with huge `MediaData` size. We can not add start field. In my current configuration it is already 440 bytes size.

Reply:
I see